docs/references/react/use-auth.mdx

```diff
@@ -9,51 +9,90 @@ The `useAuth()` hook is a convenient way to access the current auth state. This
-    };
-
-    return <div>...</div>;
-  }
-
-  ```
-
-  ```tsx filename="external-data-page.tsx"
-  import { useAuth } from '@clerk/clerk-react';
-
-  export default function ExternalDataPage() {
-    const { getToken, isLoaded, isSignedIn } = useAuth();
-
-    if (!isLoaded || !isSignedIn) {
-      // You can handle the loading or signed state separately
-      return null;
-    }
-
-    const fetchDataFromExternalResource = async () => {
-      const token = await getToken();
-      // fetch your data
-      return data;
-    }
-
-    return <div>...</div>;
-  }
-
-  ```
-</CodeBlockTabs>
+<Tabs items={['Next.js', 'React']}>
+  <Tab>
+    ### Next.js
+    <CodeBlockTabs options={["App Router", "Page Router"]}>
+      ```tsx filename="/app/pageWithData/clientComponent.tsx"
+      "use client";
+      import { useAuth } from '@clerk/nextjs';
+
+      export default function ExternalDataPage() {
+        const [data, setData] = useState<object>({});
+        const { getToken, isLoaded, isSignedIn } = useAuth();
+
+        // Fetch data from a Next.js API route handler
+        useEffect(() => {
+          if (!isLoaded || !isSignedIn) {
+            // You can handle the loading or signed state separately
+            return;
+          }
+
+          // Fetch data without async/await since this is a client component
+          getToken()
+          .then((token) =>
+          fetch('/api/protectedNextRoute', {
+            headers: {Authentication: `Bearer ${token}`}
+          }))
+          .then(response => response.json())
+                .then(data => {
+                  setData(data.data)
+                });
+        }, []);
+
+        return <div>...</div>;
+      }
+      ```
+
+      ```tsx filename="pages/index.tsx"
+      import { useAuth } from '@clerk/nextjs';
+
+      export default function ExternalDataPage() {
+        const { getToken, isLoaded, isSignedIn } = useAuth();
+
+        if (!isLoaded || !isSignedIn) {
+          // You can handle the loading or signed state separately
+          return null;
+        }
+
+        const fetchDataFromExternalResource = async () => {
+          const token = await getToken();
+          // fetch your data
+          return data;
+        };
+
+        return <div>...</div>;
+      }
+      ```
+    </CodeBlockTabs>
+  </Tab>
+
+  <Tab>
+    ### React
+    <CodeBlockTabs>
+      ```tsx filename="external-data-page.tsx"
+      import { useAuth } from '@clerk/clerk-react';
+
+      export default function ExternalDataPage() {
+        const { getToken, isLoaded, isSignedIn } = useAuth();
+
+        if (!isLoaded || !isSignedIn) {
+          // You can handle the loading or signed state separately
+          return null;
+        }
+
+        const fetchDataFromExternalResource = async () => {
+          const token = await getToken();
+          // fetch your data
+          return data;
+        }
+
+        return <div>...</div>;
+      }
+      ```
+    </CodeBlockTabs>
```

Comment:
No need for a codeblocktabs here.. this code example is already in a `<Tab />`

```suggestion
      ```tsx filename="external-data-page.tsx"
      import { useAuth } from '@clerk/clerk-react';

      export default function ExternalDataPage() {
        const { getToken, isLoaded, isSignedIn } = useAuth();

        if (!isLoaded || !isSignedIn) {
          // You can handle the loading or signed state separately
          return null;
        }

        const fetchDataFromExternalResource = async () => {
          const token = await getToken();
          // fetch your data
          return data;
        }

        return <div>...</div>;
      }
      ```
```